packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -287,6 +289,8 @@ class AnalyzeCommand extends FlutterCommand {
     server.onAnalyzing.listen((bool isAnalyzing) => _handleAnalysisStatus(server, isAnalyzing));
     server.onErrors.listen(_handleAnalysisErrors);
 
+    Cache.unlock();
```

Comment:
try / finally ?  What happens if there's an exception and we forget to unlock the file?